code/datums/elements/unfriend_attacker.dm

```diff
@@ -35,6 +35,8 @@
 	if (!isliving(attacker))
 		return
 	var/mob/living/living_attacker = attacker
+	if (!living_attacker.combat_mode && living_attacker.is_holding_item_of_type(/obj/item/pet_carrier))
```

Comment:
This feels like a very "fixes this bug and ignores like 13 other edge cases" fix.
Unfortunately I can see why it happens, as items don't require you to be in combat mode to hit people with them and a pet carrier has more than 0 force, so I don't have a very good idea about how to fix this more generically than "add a new item flag?"

Open to ideas from anyone else.

Reply:
I think the root of the issue is because the 'attack' on the corgi sends the attack signal. The weird thing for me is this signal is not sent until after the corgi is picked up by the carrier. Not really sure if messing with that could be a good solution.

Reply:
Change carriers to use `interact_with_atom` and we can avoid attacking entirely

Reply:
Don't we still wanna attack if combat mode is on? So we would implement to check if combat mode is on then call parent or do nothing?

Reply:
Yeah we can do that with `interact_with_atom`, it's so much easier